File: algorithms/sorting/insertion-sort/InsertionSort.ts
class Sorting {
	public InsertionSort(unsortedArray: number[]): number[] {
		const n: number = unsortedArray.length;
		let j: number, key: number;

		for (let i = 1; i < n - 1; i++) {
			key = unsortedArray[i];
			j = i - 1;

			while (j >= 0 && unsortedArray[j] > key) {
				unsortedArray[j + 1] = unsortedArray[j];
				j--;
			}

			unsortedArray[j + 1] = key;
		}

		return unsortedArray;
	}
}

export { Sorting };

File: algorithms/sorting/insertion-sort/InsertionSort.dart
class Sorting {
  static List<int> InsertionSort(List<int> unsortedList) {
    int n = unsortedList.length;
    int j, key;

    for (int i = 1; i < n; i++) {
      key = unsortedList[i];
      j = i - 1;

      while (j >= 0 && unsortedList[j] > key) {
        unsortedList[j + 1] = unsortedList[j];
        j--;
      }

      unsortedList[j + 1] = key;
    }

    return unsortedList;
  }
}
